apigentools/commands/push.py

```diff
@@ -3,16 +3,68 @@
 # This product includes software developed at Datadog (https://www.datadoghq.com/).
 # Copyright 2019-Present Datadog, Inc.
 import logging
+import os
 import subprocess
-import tempfile
 import time
 
+import click
+
+from apigentools import constants
 from apigentools.commands.command import Command
-from apigentools.utils import change_cwd, get_current_commit, run_command
+from apigentools.config import Config
+from apigentools.utils import change_cwd, get_current_commit, run_command, env_or_val
 
 log = logging.getLogger(__name__)
 
 
+@click.command()
+@click.option(
+    "--default-branch",
+    help="Default branch of client repo - if it doesn't exist, it will be created and pushed to instead of a new feature branch",
+    default=env_or_val("APIGENTOOLS_DEFAULT_PUSH_BRANCH", "master"),
+)
+@click.option(
+    "--dry-run",
+    help="Do a dry run of push (don't actually create and push new branches)",
+    is_flag=True,
+    default=False,
+)
+@click.option(
+    "--push-commit-msg",
+    help="Message to use for the commit when pushing the auto generated clients",
+    default=env_or_val("APIGENTOOLS_COMMIT_MSG", ""),
+)
+@click.option(
+    "--skip-if-no-changes",
+    help="Skip committing/pushing for all repositories where only .apigentools-info has changed",
+    is_flag=True,
+    default=env_or_val("APIGENTOOLS_SKIP_IF_NO_CHANGES", False, __type=bool),
+)
+@click.option(
+    "--git-email",
+    help="Email of the user to author git commits as. Note this will permanently"
+    "modify the local repos git config to use this author",
+    default=env_or_val("APIGENTOOLS_GIT_AUTHOR_EMAIL", None),
+)
+@click.option(
+    "--git-name",
+    help="Name of the user to author git commits as. Note this will permanently"
+    " modify the local repos git config to use this author",
+    default=env_or_val("APIGENTOOLS_GIT_AUTHOR_NAME", None),
+)
+@click.pass_obj
+def push(ctx_obj, **kwargs):
```

Comment:
`s/ctx_obj/ctx/`

Reply:
I chose `ctx_obj` here since this isn't actually the full `ctx`. Its only the `ctx.obj` attribute. By default, the pass_ctx decorator will pass ctx, but since we only need the passed arguments, I chose to use the pass_obj decorator - https://click.palletsprojects.com/en/7.x/complex/#the-first-child-command

Reply:
ok